src/ui/atoms/listItem/ListItem.tsx

```diff
@@ -3,57 +3,82 @@ import type { DeepReadonly } from 'superTypes'
 import { Typography } from '../typography/Typography'
 import './listItem.scss'
 import { isString } from 'utils'
+import classNames from 'classnames'
+import type { ListProps } from '../list/List'
 
-export type ListItemProps = {
+export type ListItemProps = Pick<ListProps, 'layout'> & {
   /**
    * The main title of the item.
    */
-  readonly title?: string
+  readonly title?: string | JSX.Element
   /**
    * Additional information about the item
    */
   readonly description?: string | JSX.Element
   /**
-   * An element that can provide visual information or allow interactions on the left side.
+   * An element that can provide visual information or allow interactions.
+   * The position of this element depends on the layout property value.
    */
-  readonly leftElement?: JSX.Element
+  readonly firstElement?: JSX.Element
   /**
-   * An element that can provide visual information or allow interactions on the right side.
+   * An element that can provide visual information or allow interactions.
+   * The position of this element depends on the layout property value.
    */
-  readonly rightElement?: JSX.Element
+  readonly lastElement?: JSX.Element
+  /**
+   * The callback function called when the list item is clicked.
+   * The `onClick` function is an event handler attached to the list item.
+   */
+  readonly onClick?: () => void
 }
 
-export const ListItem: React.FC<ListItemProps> = ({
-  title,
-  description,
-  leftElement,
-  rightElement
-}: DeepReadonly<ListItemProps>): JSX.Element => {
-  const renderDescription = (): string | JSX.Element | undefined =>
-    description && (
+type TitleProps = Required<Pick<ListItemProps, 'title'>>
+type DescriptionProps = Required<Pick<ListItemProps, 'description'>>
+
+const Title = ({ title }: DeepReadonly<TitleProps>): JSX.Element => {
+  if (isString(title)) {
+    return (
+      <div className="okp4-listitem-title">
+        <Typography color="inverted-text" fontSize="small" fontWeight="bold">
+          {title}
+        </Typography>
+      </div>
+    )
+  }
+
+  return <div className="okp4-listitem-title">{title}</div>
+}
+
+const Description = ({ description }: DeepReadonly<DescriptionProps>): JSX.Element => {
+  if (isString(description)) {
+    return (
       <div className="okp4-listitem-description">
-        {isString(description) ? (
-          <Typography as="div" color="inverted-text" fontSize="small">
-            {description}
-          </Typography>
-        ) : (
-          description
-        )}
+        <Typography color="inverted-text" fontSize="small">
+          {description}
+        </Typography>
       </div>
     )
+  }
 
+  return <div className="okp4-listitem-description">{description}</div>
+}
+export const ListItem = ({
+  title,
+  description,
+  firstElement,
+  lastElement,
+  onClick,
+  layout
```

Comment:
To avoid having `undefined` as value we should pass the default value
```suggestion
  layout = 'list'
```